### PR TITLE
Remove empty statement

### PR DESCRIPTION
**Pull request by clement26695**

### CODE REVIEW

File: packages/prettier-plugin-java/test/comments/comments-blocks-and-statements/_output.java [hunk 1]
```diff
@@ -3,7 +3,7 @@
 
   public void myFunction(int arg1) {
     try {
-      ;// Empty Statement
+      // Empty Statement
```
[the remark this answers is not in the document]
Comment (bd82) [reply packages/prettier-plugin-java/test/comments/comments-blocks-and-statements/_output.java]: I am with @Shaolans here. I think this represents several unique edge cases.
I would like to see most of those tested, perhaps even with a separate test input/output pair.

edge cases:
- empty statement with comment.
- empty statement in for/do header.
- empty statement without comment

**Comment by clement26695** [comment]
I added some tests and refactored the code to not modify the CST during processing.

There is still an issue in the tests I need to solve (issue on stable reformating)

File: packages/prettier-plugin-java/src/printers/printer-utils.js [hunk 2]
```diff
@@ -156,8 +156,44 @@ function findDeepElementInPartsArray(item, elt) {
...
+function hasTrailingComments(token) {
+  return token.trailingComments !== undefined;
+}
+
+function hasComments(token) {
+  return hasLeadingComments(token) || hasTrailingComments(token);
+}
+
+function isExplicitLambdaParameter(ctx) {
+  return (
```
Comment (bd82) [inline packages/prettier-plugin-java/src/printers/printer-utils.js]: JavaScript needs Kotlin's null safety features 😄 
https://kotlinlang.org/docs/reference/null-safety.html#safe-calls

File: packages/prettier-plugin-java/test/empty_statement/_input.java [hunk 3]
```diff
@@ -1,5 +1,27 @@
...
+  public void emptyStatementWithComment() {
+    ;;//EmptyStatement
+  }
+
+  public void simpleForWithEmptyStatement() {
+    for (;;);
+
+    for (;;)/*test*/;
+
+    for (;;);/*test*/
```
Comment (bd82) [inline packages/prettier-plugin-java/test/empty_statement/_input.java]: worth checking edge case of comments _between_ the for loop's semi colons?

Comment (clement26695) [reply packages/prettier-plugin-java/test/empty_statement/_input.java]: These semi colons are not considered as empty statements, so that's fine, IMO. But I can add some, that can not harm !
